pkg/controllers/user/helm/controller.go

```diff
@@ -347,9 +374,23 @@ func (l *Lifecycle) createAppRevision(obj *v3.App, template, notes string, faile
 func (l *Lifecycle) writeKubeConfig(obj *v3.App, tempDir string, remove bool) (string, error) {
 	var token string
 
-	userID := obj.Annotations["field.cattle.io/creatorId"]
+	isSystemProject, err := l.isInSystemProject(obj)
+	if err != nil {
+		return "", err
+	}
+
+	userID, ok := obj.Annotations["field.cattle.io/creatorId"]
```

Comment:
So, the absence of the creatorId is now a special flag that causes new conditional logic to pick a system account. Seems like a confusing back door, not even necessarily from a security perspective, just additional logic branches to understand.